src/test/groovy/sprouts/Properties_List_Spec.groovy

```diff
@@ -712,4 +712,107 @@ class Properties_List_Spec extends Specification
+        reportInfo """
+            A common use case for empty lists and nullable empty lists is inside the change delegate.
+            A delegate holds information about the change, such as the `newValues` and `oldValues`.
+            These can also be empty. For example, when a property is removed from the list, the newVals list is empty. 
+        """
+        given : 'A "Vals" and a nullable "Vals" instance with no properties.'
+            var vals = Vals.of(String.class)
+            var valsNullable = Vals.ofNullable(String.class)
+        expect : 'The `size` method returns 0 and the `isEmpty` method returns true.'
+            vals.size() == 0
+            valsNullable.size() == 0
+            vals.isEmpty()
+            valsNullable.isEmpty()
+    }
+
+    def 'The change delegate contains information about changes made to a "Vars" list by adding properties.'() {
+        reportInfo """
+            When you add a property to a "Vars" list, the change delegate contains information about the change.
+            The `newValues` of the delegate contains the added property, and the `oldValues` is always an empty list.
+        """
+        given : 'A "Vars" instance having a few properties.'
+            var vars = Vars.of("a", "d", "e")
+        and : 'We register a listener that will record the last change for us.'
+            var change = null
+            vars.onChange({ change = it })
+        when : 'We add a property to the list with the `add` method.'
+            vars.add("f")
+        then : 'The `newValues` of the change delegate should be a property list with the added property.'
+            change.newValues().size() == 1
+            change.newValues().at(0).get() == "f"
+        and : 'The `oldValues` of the change should be an empty property list.'
+            change.oldValues().isEmpty()
+        when : 'We add a property to the list using the `add` method.'
+            vars.add(Var.of("g"))
+        then : 'The `newValues` should contain the added property.'
+            change.newValues().size() == 1
+            change.newValues().at(0).get() == "g"
+        and : 'The `oldValues` method should be an empty list.'
+            change.oldValues().isEmpty()
+        when : 'We add a property to the list using the `addAt` method.'
+            vars.addAt(1, "b")
+        then : 'The `newValues` should contain the added property.'
+            change.newValues().size() == 1
+            change.newValues().at(0).get() == "b"
+        and : 'The `oldValues` method should be an empty list.'
+            change.oldValues().isEmpty()
+        when : 'We add a property to the list using the `add` method.'
+            vars.addAt(2, Var.of("c"))
+        then : 'The `newValues` should contain the added property.'
+            change.newValues().size() == 1
+            change.newValues().at(0).get() == "c"
+        and : 'The `oldValues` method should be an empty list.'
+            change.oldValues().isEmpty()
```

Comment:
Nicely written feature spec! :)
I think in this test it would also be super valuable to check
if the the `ValsDelegate::index` reflects the correct change position.